### ISSUE TITLE
Mjunoon.tv plugin works in Kodi, but not with http proxy. YouTube recorded don't stream !

### ISSUE THREAD

**Issue by siliconhippy** [streamlink/streamlink#5452]
### Checklist

- [X] This is a plugin issue and not a different kind of issue
- [X] [I have read the contribution guidelines](https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#contributing-to-streamlink)
- [X] [I have checked the list of open and recently closed plugin issues](https://github.com/streamlink/streamlink/issues?q=is%3Aissue+label%3A%22plugin+issue%22)
- [X] [I have checked the commit log of the master branch](https://github.com/streamlink/streamlink/commits/master)

### Streamlink version

Latest stable release

### Description

I have no problem streaming mjunoon.tv and live YouTube channels on kodi.tv using a streamlink GUI called TVlink developed by AlexElec.tv which is an *ELEC Linux  based fork for running Kodi on various aarch64 Amlogic tv boxes and x86-64 PCs.

When I pass through the TVlink traffic via Kodi > System>  Internet> proxy ( http), live YouTube channels through TVlink ( aka streamlink) work fine. 

But mjunoon.tv channels are blocked on this squid http proxy set up on AWS Virginia. I read that mjunoon.tv is https streamed from an Amsterdam VPS.

1. So first, does streamlink have a problem with proxies ? Or http or other specific proxies ?
2. Any workarounds if yes above?
3. Does mjunoon.tv need https for mandatory cert exchange? Or can this issue be avoided by some http header spec in streamlink or m3u URL specification ?
4. Recorded YouTube channels don't play in Kodi. Someone said this is because YouTube recorded channels are stream split into separate video and audio, and Kodi ffmpeg integration causes a big delay/non sync. Any workarounds in streamlink ?

### Debug log

```text
# ( I don't get any error messages in TVLink !)
```

**Comment by bastimeyer** [comment]
No support for any forks or downstream projects.

Also nice license violation:
https://github.com/AlexELEC/TVLINK-x86_64/tree/259716a4ada5afddcd71f9908b94cebd7755d759/libs/streamlink

**Comment by siliconhippy** [comment]
Sorry ! Had no idea about the violation status - just using it.
Will post the problem separately.... hopefully will not be struck didn't, as these two issues could be generic!

**Comment by bastimeyer** [comment]
We can only support users of Streamlink itself. Everything else is out of scope and just wastes our time.

If you think that there's an issue with Streamlink's features and/or any of its plugins, then open an issue and properly fill in all the required details.
